models/service_helpers.go

```diff
@@ -253,19 +257,14 @@ func RemoveService(q *reform.Querier, id string, mode RemoveMode) error {
 	if err != nil {
 		return errors.Wrap(err, "failed to select Agent IDs")
 	}
-	if len(structs) != 0 {
-		switch mode {
-		case RemoveRestrict:
-			return status.Errorf(codes.FailedPrecondition, "Service with ID %q has agents.", id)
-		case RemoveCascade:
-			for _, str := range structs {
-				agentID := str.(*Agent).AgentID
-				if _, err = RemoveAgent(q, agentID, RemoveCascade); err != nil {
-					return err
-				}
-			}
-		default:
-			panic(fmt.Errorf("unhandled RemoveMode %v", mode))
+	if len(structs) != 0 && mode == RemoveRestrict {
```

Comment:
IMHO, switch case looks easier to understand

Reply:
Well, current approach have flat structure and `mode` validation is in one place at the top.

Reply:
We do use `https://github.com/nishanths/exhaustive` which is helpful only if we use switches.

Reply:
so return back?

Reply:
Yes, please

Reply:
done